api/v1alpha1/driver_types.go

```diff
@@ -81,46 +84,69 @@ const (
 )
 
 type EncryptionSpec struct {
+	//+kubebuilder:validation:Required
```

Comment:
why this is required?

Reply:
As explained here: https://github.com/ceph/ceph-csi-operator/pull/62#discussion_r1694949263
I prefer us to be explicit on every field and not relay on package-level default. The main reason is that one can have a package level default in one API filed but it will effect all other APIs in the same package, which changes the validation semantics. 

Being explicit solves it